sdk/storage/Azure.Storage.Files.Shares/src/ShareClient.cs

```diff
@@ -749,11 +810,15 @@ private void SetNameFieldsIfNull()
         /// A <see cref="RequestFailedException"/> will be thrown if
         /// a failure occurs.
         /// </remarks>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+#pragma warning disable AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
         public virtual Response<bool> DeleteIfExists(
-            bool includeSnapshots = true,
-            CancellationToken cancellationToken = default) =>
+#pragma warning restore AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
+            bool includeSnapshots,
```

Comment:
Should we expose this as bool? What if they add more Delete Options?

Reply:
I think it's fine for now, if they add another optional parameter, I'll switch to `ShareDeleteOptions`.